pallets/registrar/src/lib.rs

```diff
@@ -320,6 +344,17 @@ pub mod pallet {
 
     #[pallet::hooks]
     impl<T: Config> Hooks<BlockNumberFor<T>> for Pallet<T> {
+        fn on_initialize(_n: BlockNumberFor<T>) -> Weight {
+            // Account for on_finalize weight
+            let mut weight = Weight::zero().saturating_add(T::DbWeight::get().reads(1));
+
+            let buffered_paras = BufferedParasToDeregister::<T>::get();
+            for _para_id in buffered_paras {
+                weight += T::InnerRegistrar::deregister_weight();
```

Comment:
use buffered_para.len()

Reply:
fix included in 9a76e31422088f5a1231e8671063cb0ed3da1581 as well.